src/ManifoldsBase.jl

```diff
@@ -262,6 +262,9 @@ function check_size(M::AbstractManifold, p, X)
     end
 end
 
+convert(T::Type, ::AbstractManifold, p) = convert(T, p)
```

Comment:
Could you add a bit of documentation here?

Reply:
Sure, done.